sormas-rest/src/main/java/de/symeda/sormas/rest/security/SormasRestUserRightFilter.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2022 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.symeda.sormas.rest.security;
+
+import static javax.ws.rs.Priorities.AUTHORIZATION;
+
+import java.io.IOException;
+
+import javax.annotation.Priority;
+import javax.enterprise.inject.spi.CDI;
+import javax.security.enterprise.AuthenticationStatus;
+import javax.security.enterprise.SecurityContext;
+import javax.security.enterprise.authentication.mechanism.http.AuthenticationParameters;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Response;
+
+import de.symeda.sormas.api.user.UserRight;
+
+@Priority(AUTHORIZATION)
+public class SormasRestUserRightFilter implements ContainerRequestFilter {
```

Comment:
@leventegal-she Why is this class needed? Isn't this already done by the Keycloak and SormasIdentityStore classes?

Reply:
The identity store classes do not produce the correct status code in the http response.